appinterface/projection/crossfire/crossfire.go

```diff
@@ -461,9 +484,115 @@ func (projection *Crossfire) checkTaskNetworkProposalVote(
+		phaseNumberPrefix = constants.PHASE_2_TX_SENT_PREFIX
+	} else if blockTime.After(projection.phaseThreeStartTime) && blockTime.Before(projection.competitionEndTime) {
+		phaseNumberPrefix = constants.PHASE_3_TX_SENT_PREFIX
+	}
+	for _, sender := range msgTransactionCreated.Senders {
+
+		// Only considering Pubkey address for now
+		if sender.Type == constants.TYPE_URL_PUBKEY && sender.MaybeThreshold == nil {
+
+			// Increment count for address as per PHASE
+			phaseAddressCountDbKey := phaseNumberPrefix + constants.DB_KEY_SEPARATOR + sender.Pubkeys[0]
+			errIncrementing := crossfireValidatorStatsView.IncrementOne(phaseAddressCountDbKey)
+			if errIncrementing != nil {
+				return fmt.Errorf("error Phase wise tx sent count increment: %v", errIncrementing)
+			}
+
+			// Increment TOTAL count for address
+			totalAddressCountDbKey := constants.TOTAL_TX_SENT_PREFIX + constants.DB_KEY_SEPARATOR + sender.Pubkeys[0]
+			errIncrementingTotal := crossfireValidatorStatsView.IncrementOne(totalAddressCountDbKey)
+			if errIncrementingTotal != nil {
+				return fmt.Errorf("error Incrementing tx sent count: %v", errIncrementingTotal)
+			}
+
 		}
 	}
+	return nil
+}
+
+// Compute Tx Sent Rank for participating validators
+func (projection *Crossfire) computeTxSentRank(
+	crossfireValidatorStatsView *view.CrossfireValidatorsStats,
+	crossfireValidatorView *view.CrossfireValidators,
+	blockTime utctime.UTCTime,
+) error {
+	// Using dummy list for development
+	participantsList, errGetRequest := projection.Client.Participants()
+	if errGetRequest != nil {
+		return fmt.Errorf("[error] Getting Participants list %w", errGetRequest)
+	}
+
+	// Get All Tx Count Sorted
+	dbParticipantWithCountList, errDbCount := crossfireValidatorStatsView.FindAllBy(constants.TOTAL_TX_SENT_PREFIX)
+	if errDbCount != nil {
+		return fmt.Errorf("[error] Database Participant With Total Count %w", errDbCount)
+	}
+	if len(dbParticipantWithCountList) < 1 {
+		// No Participants to Rank
+		return nil
+	}
 
+	// Sort With Most Tx Sent First
+	sort.SliceStable(dbParticipantWithCountList, func(i, j int) bool {
+		return dbParticipantWithCountList[i].Value > dbParticipantWithCountList[j].Value
+	})
+
+	var rank int = 1
+	for _, dbParticipant := range dbParticipantWithCountList {
+		dbParticipantPrimaryAddress := strings.Split(dbParticipant.Key, constants.DB_KEY_SEPARATOR)[1]
+
+		// Check for each Participant in URL
+		for _, participant := range *participantsList {
+
+			// Check if the primary address match
+			if participant.PrimaryAddress == dbParticipantPrimaryAddress {
+
+				// Check if Participant is a Registered Crossfire Validator
+				registeredCrossfireValidator, errFindingValidator := crossfireValidatorView.FindBy(view.CrossfireValidatorIdentity{
+					MaybeOperatorAddress: &participant.OperatorAddress,
+				})
+				if errFindingValidator != nil || registeredCrossfireValidator == nil {
+					break
+				}
+
+				// Update Ranks with specific Validator
```

Comment:
From the previous discussion, there is a hidden logic here, it needs to check if two tx count is the same and return the same rank. So that it can have multiple rank 10 and all of them will get the reward.

Reply:
Just need to confirm this? Whether we are going with this way or not? I am waiting on the discussion to conclude on the slack thread.   I will update accordingly thanks Owen.